src/matvis/gpu.py

```diff
@@ -443,22 +463,29 @@ def simulate(
             # E-field components, and integrate over the sky.
             # Remember cublas is in fortran order...
             # v_gpu is (nfeed * nant, nax * nsrcs_up)
-            cublas_complex_mm(
-                h,
-                "c",  # conjugate transpose for first (remember fortran order)
-                "n",  # no transpose for second.
-                nfeed * nant,
-                nfeed * nant,
-                nax * nsrcs_up,
-                1.0,
-                v_gpu.gpudata,
-                nax * nsrcs_up,
-                v_gpu.gpudata,
-                nax * nsrcs_up,
-                0.0,
-                matvis_gpus[c].gpudata,
-                nfeed * nant,
-            )
+            if use_loop_over_antpairs:
+                for j in range(nfeed):
+                    for k in range(nfeed):
```

Comment:
Does the double loop over nfeed slow this down at all?  I think with cupy the nfeed dimension could be wrapped into the dot product like it is in the cpu version.  Maybe it doesn't matter though.